src/main/java/com/example/accountz/service/UserService.java

```diff
@@ -0,0 +1,81 @@
+package com.example.accountz.service;
+
+import com.example.accountz.exception.GlobalException;
+import com.example.accountz.model.UserDto;
+import com.example.accountz.persist.entity.UserEntity;
+import com.example.accountz.persist.repository.UserRepository;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import static com.example.accountz.type.ErrorCode.*;
+
+@Slf4j
+@Service
+@AllArgsConstructor
+public class UserService implements UserDetailsService {
+
+    private final UserRepository userRepository;
+    private final PasswordEncoder passwordEncoder;
+
+    public UserEntity registerUser(UserDto.SignUp member)
+            throws GlobalException {
+        boolean exists =
+                this.userRepository.existsByEmail(
+                        member.getEmail());
+
+        if (exists) {
+            throw new GlobalException(ALREADY_REGISTERED_EMAIL);
+        }
+
+        member.setPassword(
+                this.passwordEncoder.encode(member.getPassword()));
+
+        return saveUser(member);
+
+
+    }
+
+    private UserEntity saveUser(UserDto.SignUp member) {
+
+        try {
+            return this.userRepository.save(member.toEntity());
+
+        }catch (Exception e){
+            throw new GlobalException(INTERNAL_SERVER_ERROR, e.getMessage());
+        }
```

Comment:
coding convention 관련된 내용을 한번 읽어보시고, 차근차근 습관을 들이면 좋을 것 같아요. 현업에서는 아래와 같은 style 로 작성해서 개발자마다 다른 style 을 최소화하여 혼란을 줄이기 위해 노력하고 있습니다.
https://naver.github.io/hackday-conventions-java/#sub-flow-after-brace
```suggestion
        } catch (Exception e) {
            throw new GlobalException(INTERNAL_SERVER_ERROR, e.getMessage());
        }
```

Reply:
감사합니다!